### ISSUE TITLE
Codes are inconsistent with the paper

### ISSUE THREAD

**Issue by YecanLee** [FreddeFrallan/Hyperfitting#2]
Hi! Thank you for your amazing work first.

I would like to ask several questions here: 
1. The code seems like using the `AdamW` instead of `Adam` claimed in the paper, is there any reason for this difference?
2. The code seems like used the `wikitext` for training instead of the `Fiction` dataset claimed in the paper, is this just a typo? 
3. I trained on the `Fiction` dataset for `20` epochs with the optimizers and lr claimed in the paper on TinyLlama model, the loss is not converged to zero yet, I guess because I used gradient clip which the original codes did not use, but I am wondering if the `Figure2` in the paper was actually done on the `wikitext` dataset? . 

Thank you for your time in advance!

**Comment by FreddeFrallan** [comment]
Hey there,
Glad you find this wonky phenomenon interesting.

Since the “code” required for Hyperfitting is so standardized, meaning it’s just regular fine-tuning, we did not really put too much stuff in this repo (as you can see). The code you see here is therefore just an example of how to achieve it, and not what we used during experimentation for the paper.

**In regards to the dataset:**
In the paper we used the datasets that we specified, but from what we have observed it seems to pretty much happen for any reasonable dataset. The only time it did not, was when we introduced heavy amounts of noise into the data.


**In regards to the optimizer:**
Again, what is specified in the paper is what we used for those experiments. However, I can also share that Hyperfitting is seemingly independent of the optimizer as well. From what we have observed, as long as you somehow manage to get the training loss to virtually zero, this phenomenon seems to occur. Pretty sure we even managed to get SGD to Hyperfitt after a lot of lr-tuning, for some particular dataset. (We even did a big thing where we rotated between different optimizers which we also got to succeed)

Good luck, I hope that you find some insights to this phenomenon.

**Comment by YecanLee** [comment]
Thank you for your super quick reply and as well as shared lots of insights with me! I think this is definitely a super interesting phenomenon and definitely worth to be explored more, I would like to share with the research community if I could find something interesting later 😄
